app/components/publications/publication-status-selector.js

```diff
@@ -0,0 +1,64 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class PublicationStatusSelector extends Component {
+  @service store;
+
+  @tracked publicationStatusses
+  @tracked statusOptions = [];
+
+  constructor() {
+    super(...arguments);
+    this.publicationStatusses = this.store.peekAll('publication-status').sortBy('position');
+    this.fillStatusOptions();
+  }
+
+  fillStatusOptions() {
+    for (const publicationStatus of this.publicationStatusses) {
+      let icon;
+      if (publicationStatus.isPending) {
+        icon =  {
+          svg: 'clock',
+          color: 'warning',
+        };
+      } else if (publicationStatus.isPublished) {
+        icon =   {
+          svg: 'circle-check',
+          color: 'success',
+        };
+      } else if (publicationStatus.isPaused) {
+        icon =  {
+          svg: 'circle-pause',
+          color: 'muted',
+        };
+      } else if (publicationStatus.isWithdrawn) {
+        icon =   {
+          svg: 'circle-error',
+          color: 'danger',
+        };
+      } else {
+        icon =  {
+          color: 'muted',
+        };
+      }
+      const option = {
+        id: publicationStatus.id,
```

Comment:
Ik zou er waarschijnlijk voor gekozen hebben van  het hele `status` object mee te geven ipv deze 2 properties "uit te pakken". Dat zou de vergelijking in de getter hieronder ook eenvoudiger maken. Detaillistisch ... voor mij mag het blijven zoals het is.

Reply:
Ik sluit me daar wel bij aan :innocent:

Reply:
Ik snap waarom jullie dit zeggen, maar ik merk op dat AuPowerSelect een id verwacht in zijn @options

Reply:
@erikap Als je https://github.com/kanselarij-vlaanderen/kaleidos-frontend/blob/development/app/components/web-components/au-power-select.hbs bekijkt, wordt het snel duidelijk waarom Tanee dit pad gekozen heeft. De `AuPowerSelect` zoals nu geïmplementeerd, moedigt meer dan wenselijk een nauwe koppeling tussen data en UI aan. Ik heb het even in beraad genomen om elders naar een betere aanpak te zoeken, maar graag ook jou feedback.

Reply:
Hier alvast één andere mogelijk aanpak die ik in het verleden toegepast heb: https://github.com/kanselarij-vlaanderen/kaleidos-frontend/blob/development/app/components/decision/decision-result-select.hbs

Reply:
Bij nader inzien lijkt `AuPowerSelect` me niet de juiste abstractie te zijn. Het lijkt me best dat we `AuPowerSelect` afslanken of gehaal verwijderen. Meeste customizatie van PowerSelect gebeurt toch op CSS niveau. Voor custom selectors met iconen voorzien we aparte ember-componenten. Indien we dan later in deze custom "icoon-selectors" een grootst gemene deler vinden, kunnen we nog steeds een generiekere component maken.  
@TaneeG Je gebruikt hier dus best `PowerSelect` ipv `AuPowerSelect`. Dat zal je wél toelaten om zelf het property path naar icoon en label te specifiëren.